sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        self.logger.info(f'Successfully got {len(response.json()[API_FIELD_RESULTS])} documents')
+        return response.json()[API_FIELD_RESULTS]
+
+    def delete_documents(self, document_ids: list = None):
+        """
+        Deletes a list of documents or all documents
+        :param document_ids: (optional) A list of document IDs that shall be deleted. If none is given, all documents
+        are deleted.
+        :return: The API endpoint response as dictionary
+        """
+        payload = {API_FIELD_VALUE: document_ids} if (document_ids is not None) else {}
+
+        response = self.delete(DOCUMENT_ENDPOINT, json=payload,
+                               log_msg_before=f"Deleting {len(document_ids if payload else 'all')} documents",
+                               log_msg_after='Successfully deleted documents')
+        return response.json()
+
+    def get_document_result(self, document_id, extracted_values: bool = None, return_null_values: bool = False):
+        """
+        Gets the processing result of an uploaded document by document ID. Raises en exception, when the documents
+        failed or didn't finish processing after the maximum number of requests
+        :param document_id: The ID of the document uploaded for processing
+        :param extracted_values: (optional) Flag if the extracted values or the ground truth should be returned. If set
+        to `True` the extracted values are returned. If set to `False` the ground truth is returned. If no ground truth
+        is available, the extracted values will be returned either way. If `None` is given, the ground truth is returned
+        if available
+        :param return_null_values: Flag if fields with null as value should be included in the response or not.
+        Default is False
+        :return: The result of the processed document or the ground truth as dictionary
+        """
+        params = {
+            API_FIELD_RETURN_NULL: return_null_values
+        }
+        if extracted_values is not None:
+            params[API_FIELD_EXTRACTED_VALUES] = extracted_values
+
+        response = self._poll_for_url(DOCUMENT_ID_ENDPOINT.format(document_id=document_id), params=params,
+                                      log_msg_before=f'Getting document result with ID {document_id}',
+                                      log_msg_after=f'Successfully got document result with ID {document_id}')
+        return response.json()
+
+    def upload_enrichment_data(self, client_id, data, data_type, subtype=None):
+        """
+        Creates one or more enrichment data records. The function returns after all data was created successfully or
+        raises an exception if something went wrong.
+        :param client_id: The client ID for which the data records shall be created.
+        :param data: A list of data to be uploaded. A 'businessEntity' record has the following format:
```

Comment:
Let's document this format in the help portal and provide a link to it.

Reply:
We can link this side: https://help.sap.com/viewer/5fa7265b9ff64d73bac7cec61ee55ae6/SHIP/en-US/d3c3be59e6a0405d9b759a66c859da87.html

Reply:
Is the reference to the documentation enough as it is now?